### ISSUE TITLE
Steam Flatpak doesn't work at all

### ISSUE THREAD

**Issue by Melechtna** [crschnick/pdx_unlimiter#115]
~/.var/app/com.valvesoftware.Steam/data/Steam/steamapps/common/Stellaris

~/.var/app/com.valvesoftware.Steam/.local/share/Paradox Interactive/Stellaris/save games/

Please fix this

**Comment by crschnick** [comment]
what do you mean by does not work? It does not recognize the installation when you set `/home/?/.var/app/com.valvesoftware.Steam/data/Steam/steamapps/common/Stellaris` in the settings menu? It's best to try without the tilde here.

**Comment by Melechtna** [comment]
> what do you mean by does not work? It does not recognize the installation when you set `/home/?/.var/app/com.valvesoftware.Steam/data/Steam/steamapps/common/Stellaris` in the settings menu? It's best to try without the tilde here.

Correct, because it keeps trying to look in ~/.local/share/Paradox Interactive/Stellaris/, which is only valid when steam isn't a flatpak

**Comment by crschnick** [comment]
Oh I see. Have you tried creating a symlink? That might solve your issue in the short term until I find some time to see how to fix this

**Comment by Melechtna** [comment]
> Oh I see. Have you tried creating a symlink? That might solve your issue in the short term until I find some time to see how to fix this

Was the first thing I attempted, flatpaks have special permissions set to prevent external system access that it isn't granted, which seems to be messing with the symlinks as they don't even work for navigation.

**Comment by crschnick** [comment]
Would pdxu then even be able to access that data if it would look in the right location? You can experiment with flatseal to open the file system for this.

**Comment by Melechtna** [comment]
> Would pdxu then even be able to access that data if it would look in the right location? You can experiment with flatseal to open the file system for this.

If it's pointed DIRECTLY at it, then yes, softlinks are a little more finiky.

**Comment by crschnick** [comment]
So in theory you should be able fix the symlink issue with https://github.com/tchx84/Flatseal/blob/master/DOCUMENTATION.md#filesystem

**Comment by Melechtna** [comment]
> So in theory you should be able fix the symlink issue with https://github.com/tchx84/Flatseal/blob/master/DOCUMENTATION.md#filesystem

I've already got these permissions in place, but unless I start mucking with the ACTUAL FS permissions flags (bad idea) symlinks won't function correctly, might break the flatpak, or give it far more permissions than it should be granted. I shouldn't need to chmod the whole flatpak to make this program work, it should recognize we're loading from .var, and shift where it's looking.

**Comment by crschnick** [comment]
Yeah there the are a couple of things that would have to be adapted to work with flatpak steam paths, it's not just the user data. There's also the steam workshop directories and more. so I'm not sure when I will maybe find the time to fix that.

If anyone wants to contribute a fix for this, I would appreciate it. Also because I don't have a Linux system that can run any games right now.

**Comment by Melechtna** [comment]
> Yeah there the are a couple of things that would have to be adapted to work with flatpak steam paths, it's not just the user data. There's also the steam workshop directories and more. so I'm not sure when I will maybe find the time to fix that.
> 
> If anyone wants to contribute a fix for this, I would appreciate it. Also because I don't have a Linux system that can run any games right now.

Not REALLY, flatpaks always go in ~/.var/app/application.whatever.whatever

In the case of steam, I've pointed where steam is, once you're in the Steam folder, everything follows standard convention, I.E. the workshop files are still in the Steam folder, wherever they usually were (I forget at this moment)

**Comment by crschnick** [comment]
Yeah I was talking about having to adapt the path handling in multiple locations in pdxu to use the flatpak paths, e.g. in the workshop support.

It was honestly not initially designed for flatpak, the paths are somewhat hardcoded. If someone wants to contribute, they can update this to be more dynamic and support flatpak. But that would also involve testing whether everything works correctly, which I for example can't do.
